streaming-http-channel-2023.bs

```diff
@@ -0,0 +1,213 @@
+
+: receiveFrom
+:: The [receiveFrom](https://solid.github.io/notifications/protocol#notify-receiveFrom) property
+    is used in the body of the subscription response.
+    The value of source property MUST be a URI, using the `https` scheme.
+
+A client establishes a subscription using the StreamingHTTPChannel2023 type by sending an authenticated
+subscription request to the Subscription Resource retrieved via Solid Notifications discovery.
+
+For StreamingHTTPChannel2023 interactions, the client sends a JSON-LD payload to the appropriate
+Subscription Resource via POST. The only required fields in this interaction are type and topic.
+The type field MUST contain the type of channel being requested: `StreamingHTTPChannel2023`
+The topic field MUST contain the URL of the resource a client wishes to subscribe to changes.
+
+<script type="text/turtle">
+PREFIX doap: <http://usefulinc.com/ns/doap#>
+PREFIX spec: <http://www.w3.org/ns/spec#>
+
+<>
+  a doap:Specification ;
+  spec:requirement
+        <#req1> ,
+        <#req2> ,
+        <#req3> .
+<#req1>
+  spec:requirementSubject <#subscription-request> ;
+  spec:requirementLevel spec:MUST ;
+  spec:statement "The type field MUST contain the type of channel being requested."@en .
+<#req2>
+  spec:requirementSubject <#subscription-request> ;
+  spec:requirementLevel spec:MUST ;
+  spec:statement "The topic field MUST contain the URL of the resource a client wishes to subscribe to changes."@en .
+<#req3>
+  spec:requirementSubject <#subscription-response> ;
+  spec:requirementLevel spec:MUST ;
+  spec:statement "The value of source property MUST be a URI, using the https scheme."@en .
+
+</script>
+
+## Subscription Example ## {#example}
+
+*This section is non-normative.*
+
+An example `POST` request using a `DPoP` bound access token is below:
+
+```http
+POST /subscription
+Authorization: DPoP <token>
+DPoP: <proof>
+Content-Type: application/ld+json
+```
+```jsonld
+{
+  "@context": ["https://www.w3.org/ns/solid/notification/v1"],
+  "type": "StreamingHTTPChannel2023",
+  "topic": "https://storage.example/resource",
+  "state": "opaque-state",
+  "expiration": "2023-12-23T12:37:15Z",
+  "rate": "PT10s"
+}
+```
+> POST request including type and topic targeting the Notification Subscription Resource.
+
+A successful response will contain a URL to the subscription Resource that can be used directly with a JavaScript client.
+
+```http
+Content-Type: application/ld+json
+```
+```jsonld
+{
+  "@context": "https://www.w3.org/ns/solid/notification/v1",
+  "type": "StreamingHTTPChannel2023",
+  "receiveFrom": "https://fetch.example/aca3cb0a-fb0a-4091-b9f8-8117d2cdb392"
+}
+```
+> Response to the POST request, including channel type and the receiveFrom URL.
+
+In JavaScript, a client can use the data in the response to establish a connection to the Fetch API.
+And define how to handle notifications
+
+```js
+const response = await fetch('https://fetch.example/aca3cb0a-fb0a-4091-b9f8-8117d2cdb392')
```

Comment:
In WebSocket, there is url, protocol parameters where the protocol is the IRI of the channel type. Do we need something along those lines for fetch? Or is the specific receiveFrom URL is intended to be unique for the channel, i.e., channel type need not be part of the request.

Reply:
I assume `receiveFrom` to be unique for the channel. Probably this should be made explicit in this spec.